sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3086,8 +3092,13 @@ def _execute_task_one_node(self, handle: ResourceHandle,
             run_fn_name = task.run.__name__
             codegen.register_run_fn(run_fn_code, run_fn_name)
 
-        job_run_id = common_utils.get_global_job_id(
-            self.run_timestamp, cluster_name=handle.cluster_name, job_id=job_id)
+        if task.spot_task is None:
+            # Don't set JOB_ID_ENV_VAR if it is a spot controller task.
+            job_run_id = task.envs.get(
```

Comment:
I didn't read the previous PR, and I'm surprised to see our internal var is getting passed in in user-level `task.envs`. We can consider refactoring this in the future.

Reply:
Yes, it can be a bit confusing. The way we can understand it is that the task holding by the spot controller is not the original task requested by the user but a internal task. For example, during the recovery of the spot job, we also change the resources in the task, though it is not requested by the user.